python-client/giskard/testing/tests/statistic.py

```diff
@@ -296,9 +286,155 @@ def test_disparate_impact(
+
+
+@test(name="Mutual Information", tags=["statistic", "classification"])
+def test_mutual_information(
+    model: BaseModel, dataset: Dataset, slicing_function: SlicingFunction, threshold: float = 0.5, debug: bool = False
+) -> TestResult:
+    """
+    TBF
+    :param model:
+    :param dataset:
+    :param slicing_function:
+    :param threshold:
+    :param debug:
+    :return:
+    """
+    import pandas as pd
+
+    sliced_dataset = dataset.slice(slicing_function)
+    check_slice_not_empty(sliced_dataset=sliced_dataset, dataset_name="dataset", test_name="test_mutual_information")
+
+    dx = pd.DataFrame(
+        {
+            "slice": dataset.df.index.isin(sliced_dataset.df.index).astype(int),
+            "prediction": model.predict(dataset).prediction,
+        },
+        index=dataset.df.index,
+    )
+    dx.dropna(inplace=True)
+
+    metric = _mutual_information(dx.slice, dx.prediction)
+    passed = metric < threshold
+
+    # --- debug ---
+    output_ds = None
+    if not passed and debug:
+        output_ds = sliced_dataset.copy()  # copy all properties
+        test_name = inspect.stack()[0][3]
+        output_ds.name = debug_prefix + test_name
+    # ---
+
+    messages = [TestMessage(type=TestMessageLevel.INFO, text=f"metric = {metric}, threshold = {threshold}")]
+
+    return TestResult(metric=metric, passed=passed, messages=messages, output_df=output_ds)
+
+
+def _theil_u(x, y):
+    import pandas as pd
+    from sklearn.metrics import mutual_info_score
+    from scipy import stats
+
+    return mutual_info_score(x, y) / stats.entropy(pd.Series(y).value_counts(normalize=True))
+
+
+@test(name="Theil's U", tags=["statistic", "classification"])
+def test_theil_u(
+    model: BaseModel, dataset: Dataset, slicing_function: SlicingFunction, threshold: float = 0.5, debug: bool = False
+) -> TestResult:
+    """
+    TBF
+    :param model:
+    :param dataset:
+    :param slicing_function:
+    :param threshold:
+    :param debug:
+    :return:
+    """
+    import pandas as pd
+
+    sliced_dataset = dataset.slice(slicing_function)
+    check_slice_not_empty(sliced_dataset=sliced_dataset, dataset_name="dataset", test_name="test_theil_u")
+
+    dx = pd.DataFrame(
+        {
+            "slice": dataset.df.index.isin(sliced_dataset.df.index).astype(int),
+            "prediction": model.predict(dataset).prediction,
+        },
+        index=dataset.df.index,
+    )
+    dx.dropna(inplace=True)
+
+    metric = _theil_u(dx.slice, dx.prediction)
+    passed = metric < threshold
+
+    # --- debug ---
+    output_ds = None
+    if not passed and debug:
+        output_ds = sliced_dataset.copy()  # copy all properties
+        test_name = inspect.stack()[0][3]
+        output_ds.name = debug_prefix + test_name
+    # ---
+
+    messages = [TestMessage(type=TestMessageLevel.INFO, text=f"metric = {metric}, threshold = {threshold}")]
+
+    return TestResult(metric=metric, passed=passed, messages=messages, output_df=output_ds)
```

Comment:
The only difference with the previous test is that we call `_teil_u` instead of `_mutual_information`. This could be refactored in a single `test_nominal_association` with a `method` attribute for example (e.g. theil or mutual_info) to avoid code repetition.